src/payment/strategies/paypal-commerce/paypal-commerce-credit-card-payment-strategy.ts

```diff
@@ -0,0 +1,91 @@
+import { Cart } from '../../../cart';
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { DisableFundingType, PaypalCommerceHostedForm, PaypalCommerceInitializationData, PaypalCommerceScriptLoader, PaypalCommerceScriptOptions } from './index';
+
+export default class PaypalCommerceCreditCardPaymentStrategy implements PaymentStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _paypalCommerceHostedForm: PaypalCommerceHostedForm,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(options.methodId));
+        const { initializationData } = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+
+        const cart = state.cart.getCartOrThrow();
+        const paramsScript = {
+            options: this._getOptionsScript(initializationData, cart),
+            attr: { clientToken: initializationData.clientToken },
+        };
+
+        const paypal = await this._paypalScriptLoader.loadPaypalCommerce(paramsScript, true);
+
+        if (options.paypalcommerce?.form) {
```

Comment:
I think it might be better if you can exit early if `paypalcommerce.form` is not passed in. This is because you don't need to make a call to fetch the client token and load the script if you can't initialise the form because the option is not passed in.

Reply:
I agree. Thank you!